parquet-variant/src/main/java/org/apache/parquet/variant/VariantBuilder.java

```diff
@@ -101,6 +135,34 @@ public Variant build() {
     return new Variant(Arrays.copyOfRange(writeBuffer, 0, writePos), metadata);
   }
 
+  // This is used in the shredded reader to check if anything has been written to the Variant since the last call.
+  // Tracking in the reader code is a bit awkward, but maybe better than polluting this interface.
+  int getWritePos() {
+    return writePos;
+  }
+
+  /**
+   * @return the constructed Variant value binary, without metadata.
+   */
+  public byte[] valueWithoutMetadata() {
+    return Arrays.copyOfRange(writeBuffer, 0, writePos);
+  }
+
+  /**
+   * Directly append a Variant value. Its keys must already be in the metadata
+   * dictionary.
+   */
+  void shallowAppendVariant(ByteBuffer value) {
```

Comment:
I'm not sure that it is obvious what "shallow" means here. If the variant is a nested type (object or array) then what happens? Instead I think this should be `appendEncodedValue`